### PR TITLE
feat: serialize system requirements

### PR DESCRIPTION
**Pull request by ruben-arts**
This allows us to add it to `pixi info` which I did:
```
        Environment: img2txt
           Features: img2txt, img2txt_base
        Solve group: img2txt
           Channels: pytorch, conda-forge, bioconda, anaconda
   Dependency count: 1
       Dependencies: python
  PyPI Dependencies: paddlepaddle, layoutparser, easyocr
   Target platforms: win-64, linux-aarch64, osx-arm64, osx-64, linux-64
System requirements: cuda = "12"
                     libc = "2.27"
```            
Printing only if it is defined by the user

### CODE REVIEW

File: crates/pixi_manifest/src/system_requirements.rs [hunk 1]
```diff
@@ -471,4 +508,61 @@ mod tests {
...
+        macos = "10.15"
+        linux = "5.11"
+        cuda = "12.2"
+        libc = "2.12"
+        archspec = "x86_64"
+        "#;
+        assert_eq!(
+            serialized.replace("\n", "").replace(" ", ""),
+            expected.replace("\n", "").replace(" ", "")
+        );
```
Comment (Hofer-Julian) [inline crates/pixi_manifest/src/system_requirements.rs:536]: Let's use snapshots here

File: crates/pixi_manifest/src/system_requirements.rs [hunk 2]
```diff
@@ -471,4 +508,61 @@ mod tests {
...
+        archspec = "x86_64"
+
+        [libc]
+        family = "glibc"
+        version = "2.12"
+        "#;
+        assert_eq!(
+            serialized.replace("\n", "").replace(" ", ""),
+            expected.replace("\n", "").replace(" ", "")
+        );
```
Comment (Hofer-Julian) [inline crates/pixi_manifest/src/system_requirements.rs:566]: Let's use shapshots here as well

**Review by Hofer-Julian — CHANGES_REQUESTED** [review]
Looks great!

File: crates/pixi_manifest/src/system_requirements.rs (hunk 1 above)
Comment (ruben-arts) [reply crates/pixi_manifest/src/system_requirements.rs:536]: Done

File: crates/pixi_manifest/src/system_requirements.rs (hunk 2 above)
Comment (ruben-arts) [reply crates/pixi_manifest/src/system_requirements.rs:566]: Done

**Review by Hofer-Julian — APPROVED** [review]

**Pull request merged**